compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -1116,7 +1116,8 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
                             .and_then(|partial_res| partial_res.full_res())
                         {
                             if !res.matches_ns(Namespace::TypeNS)
-                                && path.is_potential_trivial_const_arg()
+                            // FIXME: should this only allow single-segment paths?
+                                && path.is_potential_trivial_const_arg(self.tcx.features().min_generic_const_args())
```

Comment:
```suggestion
                                && path.is_potential_trivial_const_arg(false)
```
Yes I think this should only work for mcg paths rn. Generally this needs to be kept in sync with the logic in name resolution that attempts to resolve type args in both valuens and typens and that currently uses `false`.